pkgs/by-name/mo/moneydance/package.nix

```diff
@@ -0,0 +1,58 @@
+{ lib, stdenv, fetchurl, makeWrapper, dpkg, openjdk21, openjfx21, }:
+let
+  version = "2023.3_5064";
+  hash = "sha256-t1jBZJ2sL0YDx9915UWgpCzKnpNq1Qz5i7ygGXDI5dk=";
+in stdenv.mkDerivation rec {
+  pname = "moneydance";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://infinitekind.com/stabledl/${version}/moneydance_linux_amd64.deb";
+    inherit hash;
+  };
+
+  jdk = openjdk21.override {
+    headless = false;
+    enableJavaFX = true;
+    openjfx = openjfx21;
+  };
+
+  nativeBuildInputs = [ dpkg ];
+  buildInputs = [ jdk makeWrapper ];
+
+  dontConfigure = true;
+  dontUnpack = true;
+  dontBuild = true;
+  dontFixup = true;
+
+  installPhase = let
+    flags = lib.strings.escapeShellArgs [
+      "-client"
+      "--add-modules"
+      "javafx.swing,javafx.controls,javafx.graphics"
+      "-classpath"
+      "${placeholder "out"}/libexec/*"
+      "Moneydance"
+    ];
+  in ''
+    runHook preInstall
+
+    mkdir -p $out $out/bin
+    dpkg -x $src $out
+    mv $out/opt/Moneydance/lib $out/libexec
+    rm -rf $out/opt
+    makeWrapper ${jdk}/bin/java $out/bin/moneydance --add-flags "${flags}"
```

Comment:
I don't know that we can rely on `escapeShellArgs` working correctly inside a double-quoted context.

Perhaps add multiple `--add-flags` parameters, one for each flag.

Something like this perhaps:
```nix
lib.flip lib.concatMapStrings flags (flag: "--add flags ${lib.escapeShellArg arg} ") 
```

Reply:
Thanks @ambroisie. Experimentally, the thing I did _does_ work, but I actually like your version better for its clarity. What I did relies on rather specific and non-obvious behavior of makeWrapper.

While I've got you, can you help me figure out the error in https://gist.github.com/GrahamcOfBorg/5bd5b2c7f7b233bd83a32bab92efcf12 ? I assume my override is broken somehow, but I'm stumped. `nix-build -A moneydance` works locally. Does this mean my override trick is incompatible with `pkgs/by-name`?

Reply:
Not sure, this looks to be something specific to the JDK handling, perhaps go ask about it in Matrix.

Reply:
Looks like you could remove `headless = false`, as it's a work-around and is false by default.

```nix
# TODO(@sternenseemann): gtk3 fails to evaluate in pkgsCross.ghcjs.buildPackages
# which should be fixable, this is a no-rebuild workaround for GHC.
, headless ? stdenv.targetPlatform.isGhcjs
```

Reply:
You're right, overriding with the `enableJavaFX` argument worked through OfBorg's eval, but the others didn't, even though local builds worked in all configurations. Hm, must be something specific to how the Java packages are built.